draft-ietf-taps-arch.md

```diff
@@ -126,32 +126,37 @@ they appear in all capitals, as shown here.
 The traditional model of using sockets for networking can be represented as follows:
 
 - Applications create connections and transfer data using the socket API.
-- The socket API provides the interface to the implementations of TCP and UDP (typically implemented in the system's kernel).
-- TCP and UDP in the kernel send and receive data over the available network layer interfaces.
+- The socket API provides the interface to the implementations of TCP and UDP
+  (typically implemented in the system's kernel).
+- TCP and UDP in the kernel send and receive data over the available network
+  layer interfaces.
+- Sockets are bound directly to transport-layer and network-layer addresses,
+  obtained via a separate resolution step, usually performed by a
+  system-provided stub resolver.
 
 ~~~~~~~~~~
 
 +-----------------------------------------------------+
 |                    Application                      |
 +-----------------------------------------------------+
-           |                             |
-+---------------------+       +-----------------------+
-|  Socket Stream API  |       |  Socket Datagram API  |
-+---------------------+       +-----------------------+
-           |                             |
-+-----------------------------------------------------+
-|         TCP                           UDP           |
-|           Kernel Protocol Implementation            |
-+-----------------------------------------------------+
-                          |
+        |                 |                  |
+  +------------+     +------------+    +--------------+
+  |    stub    |     | Stream API |    | Datagram API |
+  |  resolver  |     +------------+    +--------------+
+  +------------+          |                  |
+                    +---------------------------------+
+                    |    TCP                UDP       |
+                    |    Kernel Networking Stack      |
+                    +---------------------------------+
+                                    |
```

Comment:
I think this is much better than the first suggestion, but is there a reason the box for the kernel networking stack does not expand also under the stub resolver? (But leaving DNS out.) This would seem most natural to me, but if the rest of you do not like that we can leave it as is.